extensions/locals.tf

```diff
@@ -12,28 +16,47 @@ locals {
     })
   }
 
-  // select environment for each service and expand config from "*"
-  services_select_env = { for k, v in var.args.services : k => merge(v, merge(lookup(v.environments, "*", {}), lookup(v.environments, var.environment, {}))) }
```

Comment:
Some changes follow to try an start improving the readability/grokkability of this code. 

Mostly change `k` and `v` to meaningful variable names even though it takes up a some more bytes.

I'm sure there is more we can do, but this helped me know what was what.